core/web/transfer_controller.go

```diff
@@ -40,9 +40,24 @@ func (tc *TransfersController) Create(c *gin.Context) {
 		return
 	}
 
+	if tr.FromAddress == utils.ZeroAddress {
+		jsonAPIError(c, http.StatusUnprocessableEntity, fmt.Errorf("invalid withdrawal source address address: %v", tr.FromAddress))
+		return
+	}
+
+	if !tr.AllowHigherAmounts {
+		balance := chain.BalanceMonitor().GetEthBalance(tr.FromAddress)
+
+		// ETH balance is less than the sent amount
+		if balance == nil || balance.Cmp(&tr.Amount) == -1 {
```

Comment:
I believe this does not consider gas price? Just raising this concern in case it is important.

Reply:
Good point, I'll check with Sam, since this whole implementation, and the previous one for that matter, was not taking that into consideration.

Reply:
Well, it's kind of impossible to solve this "perfectly". We can only do a best effort (since the local balance is not guaranteed to be up-to-date.

I'd say its reasonable to grab the current gas price and multiply it by the limit then add to total amount in calculations here.